docs/source/common/debugging.rst

```diff
@@ -133,43 +207,25 @@ See Also:
     - :func:`~pytorch_lightning.utilities.model_summary.summarize`
     - :class:`~pytorch_lightning.utilities.model_summary.ModelSummary`
 
-----------------
-
-Shorten Epochs
-==============
-
-Sometimes it's helpful to only use a percentage of your training, val, test or predict data (or a set number of batches).
-For example, you can use 20% of the training set and 1% of the validation set.
-
-On larger datasets like Imagenet, this can help you debug or test a few things faster than waiting for a full epoch.
-
-.. testcode::
-
-    # use only 10% of training data and 1% of val data
-    trainer = Trainer(limit_train_batches=0.1, limit_val_batches=0.01)
-
-    # use 10 batches of train and 5 batches of val
-    trainer = Trainer(limit_train_batches=10, limit_val_batches=5)
 
 ----------------
 
-Validation Sanity Check
-=======================
 
-Lightning runs a few steps of validation in the beginning of training.
-This avoids crashing in the validation loop sometime deep into a lengthy training loop.
+*************************************
+Debugging with Distributed Strategies
+*************************************
 
-(See: :paramref:`~pytorch_lightning.trainer.trainer.Trainer.num_sanity_val_steps`
-argument of :class:`~pytorch_lightning.trainer.trainer.Trainer`)
+DDP Debugging
+=============
 
-.. testcode::
+If you are having a hard time debugging DDP on your remote machine you can debug DDP locally on the CPU. Note that this will not provide any speed benifits.
```

Comment:
I would add hints on how to debug with DDP.
First, debug with printing the ranks print(f"RANK: {trainer.global_rank} {msg}"). IDE supports exploring multi process such as DDP SPawn.
Futhermore, you can use use breakpoint by protecting it with rank information.
```
if trainer.global_rank == 9:
      breakpoint()
...
```

Reply:
I would really like to have _rank_breakpoint(0) and multi_print(msg) (using color) to help debugging DDP. Maybe next PRs :)

Reply:
it often hangs a lot when I do this with DDP since we can't stop other processes from executing further. Curious to see how rank_breakpoint will work.

Reply:
a rank breakpoint could be implemented like this (did not test):

```py
def rank_breakpoint(rank=0):
    if global_rank != rank:
        barrier()  # all processes wait here except the one we want to debug
    else:
        # the chosen rank will set a trace, but we want the frame from where this function is called
        pdb.Pdb().set_trace(inspect.currentframe().f_back.f_back)
        # once breakpoint gets skipped, synchronize with all other workers
        barrier()
```

Reply:
thanks @awaelchli 
no-barrier was the reason why it used to hang a lot for me with DDP + pdb

I think we can add this.. once tested.